detector/cve/cve20258088/cve20258088.go

```diff
@@ -0,0 +1,331 @@
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/google/osv-scalibr/detector"
+	"github.com/google/osv-scalibr/extractor"
+	scalibrfs "github.com/google/osv-scalibr/fs"
+	"github.com/google/osv-scalibr/inventory"
+	"github.com/google/osv-scalibr/log"
+	"github.com/google/osv-scalibr/packageindex"
+	"github.com/google/osv-scalibr/plugin"
+	"github.com/ossf/osv-schema/bindings/go/osvschema"
+	"github.com/saferwall/pe"
+)
+
+const (
+	// Name of the detector.
+	Name = "cve/cve-2025-8088"
+)
+
+// Detector is a SCALIBR Detector for CVE-2025-8088.
+type Detector struct{}
+
+// New returns a detector.
+func New() detector.Detector {
+	return &Detector{}
+}
+
+// Name of the detector.
+func (Detector) Name() string { return Name }
+
+// Version of the detector.
+func (Detector) Version() int { return 0 }
+
+// Requirements of the detector.
+func (Detector) Requirements() *plugin.Capabilities {
+	return &plugin.Capabilities{OS: plugin.OSWindows, DirectFS: true, RunningSystem: true}
+}
+
+// RequiredExtractors of the detector.
+func (Detector) RequiredExtractors() []string {
+	return []string{"windows/ospackages"}
+}
+
+// DetectedFinding returns generic vulnerability information about what is detected.
+func (Detector) DetectedFinding() inventory.Finding {
+	return inventory.Finding{PackageVulns: []*inventory.PackageVuln{{
+		Vulnerability: osvschema.Vulnerability{
+			ID:      "CVE-2025-8088",
+			Summary: "WinRAR path traversal vulnerability (<7.13)",
+			Details: "Detects WinRAR installs or portables < 7.13, using robust and reliable PE analysis.",
+		},
+	}}}
+}
+
+// Scan checks for the presence of the BentoML CVE-2025-8088 vulnerability on the filesystem.
+func (d Detector) Scan(ctx context.Context, scanRoot *scalibrfs.ScanRoot, px *packageindex.PackageIndex) (inventory.Finding, error) {
+	var findings []*inventory.PackageVuln
+	// Match any PE file that could be WinRAR, RAR, UnRAR, UnRAR.dll, including renamed and portable versions
+	peRegex := regexp.MustCompile(`(?i)(winrar|rar|unrar).*?(\.exe|\.dll)$`)
+
+	// === Phase 1: Installed packages via package index ===
+	if px != nil {
+		for _, pkg := range px.GetAll() {
+			nameLower := strings.ToLower(pkg.Name)
+			if strings.Contains(nameLower, "winrar") {
+				normalizedVersion := normalizeVersion(pkg.Version)
+
+				if isAffectedVersion(normalizedVersion) {
+					log.Infof("Vulnerable WinRAR package installation found(Uninstall them): Package %s, Version: %s", pkg.Name, normalizedVersion)
+
+					affected := []osvschema.Affected{{
+						Package: osvschema.Package{
+							Name:      pkg.Name,
+							Ecosystem: "generic",
+						},
+						Ranges: []osvschema.Range{{
+							Type: "SEMVER",
+							Events: []osvschema.Event{{
+								Introduced: "0",
+							}, {
+								Fixed: "7.13",
+							}},
+						}},
+						Versions: []string{normalizedVersion},
+					}}
+
+					vuln := &inventory.PackageVuln{
+						Package: pkg,
+						Vulnerability: osvschema.Vulnerability{
```

Comment:
Return the same Vulnerability as in DetectedFinding() here. For other extractors we usually have a shared helper lib, e.g. https://github.com/google/osv-scalibr/blob/396e8cfb62cd9c0a439b46a12406d1677292e0eb/detector/cve/untested/cve20242912/cve20242912.go#L101

Reply:
This hasn't been resolved yet. I see you introduced a shared makePackageVuln func but it's not used for DetectedFinding() yet.